cm/master.py

```diff
@@ -1463,6 +1386,111 @@ def _add_data_fs(fs_name=None):
         else:
             log.error("Tried to initialize a cluster but received an unknown type: '%s'" % cluster_type)
 
+    def process_cluster_templates(self, cluster_type, pss, storage_type):
+        if self.app.config.cluster_templates:
+            cluster_template = None
+            for template in [template for template in self.app.config.cluster_templates if 'name' in template]:
+                if template['name'] == cluster_type:
+                    cluster_template = template
+            if not cluster_template or (not 'filesystem_templates' in cluster_template and cluster_type != "Test"):
+                log.warn("No filesystem templates defined for cluster type: {0}".format(cluster_type))
+            else:
+                self.process_filesystem_templates(cluster_type, pss, storage_type, cluster_template['filesystem_templates'])
+        elif self.app.config.filesystem_templates: # Legacy behavior - process filesystem_templates directly or snaps.yaml
+            self.process_filesystem_templates(cluster_type, pss, storage_type, self.app.config.filesystem_templates)
+        else:
+            if cluster_type != "Test":
```

Comment:
I agree with the general sentiment. In this case though, the else clause would only do a pass? (That is, if cluster_type == Test, I guess we don't really care whether a cluster template or filesystem template exists, so do you reckon we should just log a message or something?)